### PR TITLE
Adds the T-100Y howitzer and rebalances the mortar, renamed mortar.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
https://streamable.com/jb3gwo
https://streamable.com/cjkdva

Adds the T-100Y howitzer, a heavier, draggable version of the mortar. Marines get one roundstart (may change), the shells all cost 3, you get two shells. Its shells are....

White phos, for all your 'spotting' needs. Costs two, but you only get one shell per buy.
HE, when you wanna ruin somones day.
Incend with a LE payload. When you wanna set someone on fire.
Tanglefoot, same as unchanged 80mm mortar. Same cost as WP.

The gun itself is incredibly accurate and a distinct sound.

Mortar renamed to T-50S, and has a few changes...

Aims to make the mortar a more mobile, moveable tool with 
~~Shells are normal sized~~, are cheaper (1 for 2) 
HE/Incend nerfed overall, they're cheaper however.
Smoke/Flare unchanged, flare can be fired out of the Howitzer by space magic.
Tanglefoot nerfed, cost down. Still only one shell on buy.

Mortar code changed a bit, sounds are more customizeable, I suppose.

UPDATE 1

-Made mortar shells small, rather than normal out of suggestions, fixed a bug.

## Why It's Good For The Game
Makes long range artillery and short range artillery more defined in TGMC, and generally makes the mortar feel more like a medium range mortar rather than just artillery. 
Also it just sounds cool.

## Changelog
:cl:
add: Added new howitzer, the T-100Y
code: Mortar code is more malleable instead of snowflakey.
balance: Mortar rebalanced. It costs 1 for 2 shells, the shells are generally less dangerous, it is generally more portable, shells are small sized instead of bulky. It is now significantly easier to run it on the move, compared to its bigger cousin the howitzer. It has been slashed in price to cost less, 25 points per mortar.
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
YEAAHHH BABBYYY WOOO

https://user-images.githubusercontent.com/59080633/163674156-c548cc8e-bd7a-4432-bfc7-c98327e57437.mp4

**Comment by Losenis** [comment]
https://www.youtube.com/watch?v=GW6GSa14xXI

**Comment by Diegoflores31** [comment]
turning cm into WW2 literally . do marines really need MORE support fire than now ??

**Review by Kuro020 — APPROVED** [review]

File: code/game/objects/machinery/mortar.dm [hunk 1]
```diff
@@ -21,6 +21,18 @@
...
+	var/reload_sound = 'sound/weapons/guns/interact/mortar_reload.ogg' // Our reload sound.
+	var/fall_sound = 'sound/weapons/guns/misc/mortar_travel.ogg' //The sound the shell makes when falling.
+
+	// What shells can we use?
+	var/list/allowed_shells = list(
+		/obj/item/mortal_shell/he,
+		/obj/item/mortal_shell/incendiary,
+		/obj/item/mortal_shell/smoke,
+		/obj/item/mortal_shell/plasmaloss,
+	)
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:34]: autodoc where

File: code/game/objects/machinery/mortar.dm [hunk 2]
```diff
@@ -233,20 +249,21 @@
...
 /obj/item/mortar_kit
-	name = "\improper M402 mortar"
+	name = "\improper TA-50S mortar"
 	desc = "A manual, crew-operated mortar system intended to rain down 80mm goodness on anything it's aimed at. Needs to be set down first to fire. Use Ctrl-Click to deploy."
 	icon = 'icons/Marine/mortar.dmi'
 	icon_state = "mortar"
 
 	max_integrity = 200
 	flags_item = IS_DEPLOYABLE|DEPLOYED_WRENCH_DISASSEMBLE
+	var/deployed_item = /obj/machinery/deployable/mortar // What item are we gonna deploy?
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:259]: autodoc and wrong order

File: code/game/objects/machinery/mortar.dm [hunk 3]
```diff
@@ -258,12 +275,56 @@
...
+/obj/machinery/deployable/mortar/howitzer/attack_hand_alternate(mob/living/user)
+	if(!Adjacent(user) || user.lying_angle || user.incapacitated() || !ishuman(user))
+		return
+
+	if(!anchored)
+		anchored = TRUE
+		to_chat(user, span_warning("You have anchored the gun to the ground. It may not be moved."))
+	else
+		anchored = FALSE
+		to_chat(user, span_warning("You unanchored the gun from the gruond. It may be moved."))
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:315]: early return
this doesnt even need to be an if use 
ancored = !anchored
and ? :

File: code/game/objects/machinery/mortar.dm [hunk 4]
```diff
@@ -415,6 +539,49 @@
 	new /obj/item/encryptionkey/engi(src)
 	new /obj/item/encryptionkey/engi(src)
 	new /obj/item/binoculars/tactical/range(src)
+	new /obj/item/binoculars/tactical/range(src)
+	new /obj/item/encryptionkey/cas(src)
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:543]: why two binocs

File: code/game/objects/machinery/mortar.dm [hunk 5]
```diff
@@ -415,6 +539,49 @@
...
+	new	/obj/item/mortal_shell/howitzer/plasmaloss(src)
+	new	/obj/item/mortal_shell/howitzer/plasmaloss(src)
+	new /obj/item/mortal_shell/flare(src)
+	new /obj/item/mortal_shell/flare(src)
+	new /obj/item/mortal_shell/flare(src)
+	new /obj/item/mortal_shell/flare(src)
+	new /obj/item/encryptionkey/engi(src)
+	new /obj/item/encryptionkey/engi(src)
+	new /obj/item/binoculars/tactical/range(src)
+	new /obj/item/binoculars/tactical/range(src)
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:585]: why two binocs

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by BraveMole — APPROVED** [review]
Me when MJP's code is actually decent

File: code/game/objects/machinery/mortar.dm (hunk 1 above)
Comment (BraveMole) [reply code/game/objects/machinery/mortar.dm]: Still some autodoc issues

**Comment by tyeagg** [comment]
flare shells dont work on mortar anymore

**Comment by Pariah919** [comment]
> flare shells dont work on mortar anymore

works, made mortar shells small

**Pull request merged**